service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/ainotebook/post-startup.sh

```diff
@@ -310,6 +310,44 @@ fi
 # integration test will ensure that everything in script worked.
 sudo -u "$JUPYTER_USER" sh -c 'terra git clone --all'
 
+################################
+# Setup instance startup service
+################################
+# The following steps are performed
+# 1. Mount terra workspace resources. This command requires system user home
+#    directories to be mounted. We run the startup service after
+#    jupyter.service to meet this requirement.
+
+# Create a script to be executed by the startup service.
+TERRA_SETUP_SCRIPT="${USER_TERRA_CONFIG_DIR}/setup-environment.sh"
+cat <<EOF >${TERRA_SETUP_SCRIPT}
+#!/bin/bash
+
+# Mount terra workspace resources
+/usr/bin/terra resource mount
+EOF
+chmod +x "${TERRA_SETUP_SCRIPT}"
+
+# Create a systemd service to run the script on instance startup.
+cat <<EOF >"/etc/systemd/system/terra-setup-environment.service"
+[Unit]
+Description=Configure environment for terra
+After=jupyter.service
+
+[Service]
+ExecStart=/home/jupyter/.terra/setup-environment.sh
```

Comment:
```
ExecStart="${TERRA_SETUP_SCRIPT}"
User="${JUPYTER_USER}"
```

Shell variables will expand at the time this is written (unless you put the HereDoc delimiter "EOF" in quotes.

Reply:
Thanks Matt, appreciate the tips! Using variables and substitution instead of hardcoding now.